tests/python_package_test/test_engine.py

```diff
@@ -1507,40 +1507,235 @@ def test_get_split_value_histogram(self):
+        metrics_combination_cv_lambdarank("lambdarank", best_iter_ndcg, False, False)
+        metrics_combination_cv_lambdarank("lambdarank", best_iter_ndcg, True, False)
+        metrics_combination_cv_lambdarank("l2", best_iter_l2, False, False)
+        metrics_combination_cv_lambdarank("l2", best_iter_l2, True, False)
+        metrics_combination_cv_lambdarank(["ndcg", "l2"], best_iter_min, False, False)
+        metrics_combination_cv_lambdarank(["ndcg", "l2"], best_iter_ndcg, True, False)
+        metrics_combination_cv_lambdarank(["l2", "ndcg"], best_iter_min, False, False)
+        metrics_combination_cv_lambdarank(["l2", "ndcg"], best_iter_l2, True, False)
+
+        # composite name metrics(multi error)
+        X, y = make_classification(n_features=10, n_redundant=0, n_informative=10,
+                                   n_clusters_per_class=3, n_classes=10, random_state=71)
+        X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)
         lgb_train = lgb.Dataset(X_train, y_train)
         lgb_eval = lgb.Dataset(X_test, y_test, reference=lgb_train)
 
-        decreasing_generator = itertools.count(0, -1)
-
-        def decreasing_metric(preds, train_data):
-            return ('decreasing_metric', next(decreasing_generator), False)
-
-        def constant_metric(preds, train_data):
-            return ('constant_metric', 0.0, False)
-
-        # test that all metrics are checked (default behaviour)
-        gbm = lgb.train(params, lgb_train, num_boost_round=20, valid_sets=[lgb_eval],
-                        feval=lambda preds, train_data: [decreasing_metric(preds, train_data),
-                                                         constant_metric(preds, train_data)],
-                        early_stopping_rounds=5, verbose_eval=False)
-        self.assertEqual(gbm.best_iteration, 1)
-
-        # test that only the first metric is checked
-        gbm = lgb.train(dict(params, first_metric_only=True), lgb_train,
-                        num_boost_round=20, valid_sets=[lgb_eval],
-                        feval=lambda preds, train_data: [decreasing_metric(preds, train_data),
-                                                         constant_metric(preds, train_data)],
-                        early_stopping_rounds=5, verbose_eval=False)
-        self.assertEqual(gbm.best_iteration, 20)
-        # ... change the order of metrics
-        gbm = lgb.train(dict(params, first_metric_only=True), lgb_train,
-                        num_boost_round=20, valid_sets=[lgb_eval],
-                        feval=lambda preds, train_data: [constant_metric(preds, train_data),
-                                                         decreasing_metric(preds, train_data)],
-                        early_stopping_rounds=5, verbose_eval=False)
-        self.assertEqual(gbm.best_iteration, 1)
+        def metrics_combination_train_multi_err(metric_list, assumed_iteration, first_metric_only):
```

Comment:
It's too much! Is this test added with the aim to check composite metric name? If so, then lambdarank test above should be enough for that.